### ISSUE TITLE
haskell development nix-shell on mac os fails complaining about /nixpkgs/pkgs/os-specific/linux/systemd/default.nix:10:1

### ISSUE THREAD

**Issue by tinkyholloway** [NixOS/nixpkgs#19294]
## Issue description

Cannot start a nix-shell with some haskell packages on darwin. In the `nix-shell` expression below the inclusion of the `hakyll` packages causes the shell to fail with the below trace.  The same expression works on nixOS. I'm sorry, but I am not smart enough to know whether this is a bug or user error.

```
$ nix-shell --show-trace
error: while evaluating the attribute ‘shellHook’ of the derivation ‘testing’ at /Users/tinkyholloway/Projects/testing/shell.nix:10:3:
while evaluating ‘ghcWithPackages’ at /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/pkgs/development/haskell-modules/default.nix:83:27, called from /Users/tinkyholloway/Projects/testing/shell.nix:4:9:
while evaluating ‘withPackages’ at /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/pkgs/development/haskell-modules/default.nix:56:22, called from /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/pkgs/development/haskell-modules/default.nix:83:39:
while evaluating ‘callPackage’ at /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/pkgs/development/haskell-modules/default.nix:54:26, called from /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/pkgs/development/haskell-modules/default.nix:56:32:
while evaluating ‘callPackageWithScope’ at /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/pkgs/development/haskell-modules/default.nix:48:42, called from /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/pkgs/development/haskell-modules/default.nix:54:32:
while evaluating ‘callPackageWith’ at /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/lib/customisation.nix:93:35, called from /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/pkgs/development/haskell-modules/default.nix:48:49:
while evaluating ‘makeOverridable’ at /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/lib/customisation.nix:54:24, called from /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/lib/customisation.nix:97:8:
while evaluating anonymous function at /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/pkgs/development/haskell-modules/with-packages-wrapper.nix:1:1, called from /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/lib/customisation.nix:56:12:
while evaluating ‘closePropagation’ at /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/lib/deprecated.nix:227:22, called from /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/pkgs/development/haskell-modules/with-packages-wrapper.nix:40:57:
while evaluating ‘uniqList’ at /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/lib/deprecated.nix:158:14, called from /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/lib/deprecated.nix:227:29:
while evaluating ‘go’ at /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/lib/deprecated.nix:159:19, called from /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/lib/deprecated.nix:165:8:
while evaluating ‘innerClosePropagation’ at /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/lib/deprecated.nix:210:33, called from /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/lib/deprecated.nix:227:52:
while evaluating ‘innerClosePropagation’ at /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/lib/deprecated.nix:210:33, called from /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/lib/deprecated.nix:216:18:
while evaluating ‘innerClosePropagation’ at /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/lib/deprecated.nix:210:33, called from /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/lib/deprecated.nix:218:21:
while evaluating ‘uniqList’ at /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/lib/deprecated.nix:158:14, called from /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/lib/deprecated.nix:220:24:
while evaluating ‘go’ at /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/lib/deprecated.nix:159:19, called from /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/lib/deprecated.nix:165:8:
while evaluating ‘go’ at /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/lib/deprecated.nix:159:19, called from /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/lib/deprecated.nix:164:27:
while evaluating ‘go’ at /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/lib/deprecated.nix:159:19, called from /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/lib/deprecated.nix:164:27:
while evaluating the attribute ‘nativeBuildInputs’ of the derivation ‘hakyll-4.8.3.2’ at /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/pkgs/development/haskell-modules/generic-builder.nix:145:3:
while evaluating the attribute ‘__sandboxProfile’ of the derivation ‘util-linux-2.28.1’ at /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/pkgs/os-specific/linux/util-linux/default.nix:4:3:
while evaluating ‘unique’ at /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/lib/lists.nix:432:12, called from /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/pkgs/stdenv/generic/default.nix:224:75:
while evaluating ‘concatMap’ at /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/lib/lists.nix:79:18, called from /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/pkgs/stdenv/generic/default.nix:191:13:
while evaluating ‘chooseDevOutputs’ at /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/lib/attrsets.nix:463:22, called from /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/pkgs/stdenv/generic/default.nix:169:22:
while evaluating ‘optional’ at /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/lib/lists.nix:175:20, called from /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/pkgs/os-specific/linux/util-linux/default.nix:57:8:
while evaluating ‘callPackageWith’ at /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/lib/customisation.nix:93:35, called from /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/pkgs/top-level/all-packages.nix:11259:13:
while evaluating ‘makeOverridable’ at /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/lib/customisation.nix:54:24, called from /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/lib/customisation.nix:97:8:
while evaluating anonymous function at /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/pkgs/os-specific/linux/systemd/default.nix:1:1, called from /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/lib/customisation.nix:56:12:
assertion failed at /nix/store/i9jbllr1b2j2kl3ks87zvb5zg46496wa-nixpkgs-17.03pre93030.548bcd0/nixpkgs/pkgs/os-specific/linux/systemd/default.nix:10:1
```
### Steps to reproduce

On mac execute this `nix-shell` expression:

```
{ nixpkgs ? import <nixpkgs> {}, compiler ? "ghc7103" }:
let
  inherit (nixpkgs) pkgs;
  ghc = pkgs.haskell.packages.${compiler}.ghcWithPackages (ps: with ps; [
    base
    hakyll
  ]);
in
pkgs.stdenv.mkDerivation {
  name = "testing";
  buildInputs = [ ghc ];
  shellHook = "eval $(egrep ^export ${ghc}/bin/ghc)";
}
```
## Technical details
- System:

```
$ uname -a
Darwin Tinky-Holloways-Macbook-Air-2.local 14.5.0 Darwin Kernel Version 14.5.0: Wed Jul 29 02:26:53 PDT 2015; root:xnu-2782.40.9~1/RELEASE_X86_64 x86_64
```
- Nix version:

```
$ nix-env --version
nix-env (Nix) 1.11.4
```
- Nixpkgs version:

```
$ nix-instantiate --eval '<nixpkgs>' -A lib.nixpkgsVersion
"17.03pre93030.548bcd0"
```

**Comment by domenkozar** [comment]
Sounds like a bug in nixpkgs master.

**Comment by tinkyholloway** [comment]
Reference #19308

**Comment by TikhonJelvis** [comment]
The problem comes up because the `hakyll` package definition depends on `utillinux` for its tests:

```
"hakyll" = callPackage ({ ..., utillinux, ...}):
...
  testToolDepends = [ utillinux ];
...
```

This isn't related to issue #19308. I think I can fix it—I'll give it a try anyhow.

**Comment by LnL7** [comment]
Is this sill an issue? It should be fairly straightforward to override the test dependencies in `configuration-common.nix` for darwin

**Comment by domenkozar** [comment]
This is still present, upstream bug: https://github.com/NixOS/cabal2nix/issues/239

**Comment by domenkozar** [comment]
Actually this was fixed in c60b3e4bfc94c567db5d68eaaf6c8c1bbb9d43ee
